File: src/main/scala/demos/fp/order/anemic/OrderLineService.scala
package demos.fp.order.anemic

object OrderLineService:
    def createOrderLine(productId: String, amount: Int): OrderLine =
        if amount <= 0 || amount > 10 then throw Exception("Cannot create order line")
        else OrderLine(productId = productId, amount = amount)

    def addToOrderLine(orderLine: OrderLine, extraAmount: Int): OrderLine =
        val newAmount = orderLine.amount + extraAmount
        if extraAmount <= 0 || orderLine.amount + extraAmount > 10 then
            throw Exception("Cannot increase order line amount")
        else OrderLine(productId = orderLine.productId, amount = newAmount)

File: src/main/java/demos/oop/order/anemic/OrderLineService.java
package demos.oop.order.anemic;

class OrderLineService {

    private OrderLine orderLine;

    public OrderLineService(OrderLine orderLine) {
        this.orderLine = orderLine;
    }

    public void validate() {
        if (this.orderLine.getAmount() <= 0 || this.orderLine.getAmount() > 10) {
            throw new RuntimeException("Order line exceeds the maximum amount");
        }
    }

    public void addToOrderLine(int extraAmount) {
        int newAmount = this.orderLine.getAmount() + extraAmount;
        if (extraAmount <= 0) {
            throw new RuntimeException("Amount to increase must be positive");
        } else {
            this.orderLine.setAmount(newAmount);
            this.validate();
        }
    }

}
